YLTableViewTests/YLTableViewDataSourceTests.m

```diff
@@ -0,0 +1,92 @@
+//
+//  YLTableViewDataSourceTests.m
+//  YLTableView
+//
+//  Created by Ushhud Khalid on 12/9/15.
+//  Copyright © 2015 Yelp. All rights reserved.
+//
+
+#import "YLTableView.h"
+#import "YLTableViewBaseTestCase.h"
+#import "YLTableViewCell.h"
+#import "YLTableViewCellEstimatedRowHeight.h"
+#import "YLTableViewCellTestStub.h"
+#import "YLTableViewDataSource.h"
+#import "YLTableViewDataSourceTestStub.h"
+
+@interface YLTableViewDataSourceTests : YLTableViewBaseTestCase
+
+@property (strong, nonatomic) YLTableView *tableView;
+@property (strong, nonatomic) YLTableViewDataSourceTestStub *dataSource;
+
+@end
+
+@implementation YLTableViewDataSourceTests
+
+static int kSectionIndexTableView = 0;
+static int kRowIndexTableViewConformingCell = 0;
+static int kRowIndexTableViewNonConformingCell = 1;
+
+#pragma mark - Setup/Teardown
+
+- (void)setUp {
+  [super setUp];
+  
+  _tableView = [[YLTableView alloc] init];
+  _dataSource = [[YLTableViewDataSourceTestStub alloc] init];
+  _tableView.dataSource = _dataSource;
+  _tableView.delegate = _dataSource;
+  
+  _dataSource.tableViewCells = [self _cellsForTableView];
+}
+
+#pragma mark - Helpers
+
+- (NSDictionary<NSIndexPath *, YLTableViewCell * > *)_cellsForTableView {
+  NSMutableDictionary * dict = [[NSMutableDictionary alloc] init];
+  
+  YLTableViewCell * conformingCell = [[YLTableViewCellTestStub alloc] init];
+  YLTableViewCell * nonConformingCell = [[YLTableViewCell alloc] init];
+
+  [dict setObject:conformingCell forKey:[self _indexPathForConformingCell]];
+  [dict setObject:nonConformingCell forKey:[self _indexPathForNonConformingCell]];
```

Comment:
This dictionary doesn't need to be mutable, since you can create it with a literal:
``` objc
return @{
  [self _indexPathForConformingCell]: [[YLTableViewCellTestStub alloc] init],
  [self _indexPathForNonConformingCell]: [[YLTableViewCell alloc] init],
};
```

Yay literals!